Latest versions jun2025

emanlove opened the pull request:
Various cleanup work on the internal test suite and bringing up the versions of parts of the tech stack.

emanlove committed:
Updating selenium and robotframework versions we test against

emanlove committed:
Fixed required version syntax

emanlove committed:
Bump lower Python version tested to 3.9

emanlove committed:
Updated expected error messages on a couple tests

emanlove committed:
Trying out key value pairs on python version for more expressive matrix

emanlove committed:
Limit test matrix to latest python version and latest rf version

Also reveted back from expressive matrix options as I need to learn how
to use key-value pairs within gh actions.

emanlove committed:
Uniquely name the artifacts

emanlove committed:
Try slight delay in mouse over test

emanlove committed:
Try slight delay in mouse over error test

emanlove committed:
Try slowing down tests a bit

emanlove committed:
Trying RF verion 7.2.2

emanlove committed:
Put back in previous selenium versions

emanlove committed:
Added install chromedriver along side of chrome

emanlove committed:
Trying Chrome version 138

Also reduce test matrix for now and set setup-chrome action to v1 (as
compared to @latest)

emanlove committed:
Trying chrome version 137 ..

Looks like the system has Chrome 1.37 already installed. This will setup chrome 137
somewhere but more importantly add the driver. Realizing I may need to rework how we
setup the machine ..

emanlove committed:
Turn off Chrome password leak detector

emanlove committed:
Tweak some timing on the alert tests

emanlove committed:
Trying to get v138 chromedriver ..

emanlove committed:
Expanding out the version text matrix verifying we are still good across many versions

emanlove committed:
Add latest selenium version 4.34.2 into test matrix

aaltat commented on atest/acceptance/keywords/alerts.robot:
```diff
@@ -43,6 +43,7 @@ Handle Alert returns message
 
 Handle Alert with custom timeout
     Click Button    Slow alert
+    Sleep  0.1s
```
Doesn’t this indicate problem in the library, instead of doing sleep, should library be fixed?

emanlove replied:
Updated this. I removed the sleep but also extended the given timeout. For this test the alert was delayed 500ms and it appears that timing out at 1sec (or 500ms later) is too short to reasonable expect for alert triggered and then detected. This extension gives it a reasonable amount of time after triggering.

emanlove committed:
Removed sleep and extended expected timeout.

Previously the timeout was 1 sec and the alert was delayed 500ms. With a normal execution
it seems that that 500ms diference is just not long enough to reasonably expect the alert
to appear. In reviewing this test it was shown any delay was useful and the needed time
for an alert to triggered and then recognized would need more than 500ms.
